### PR TITLE
Require Node.js 14.16

### PR DESCRIPTION
**Pull request by Richienb**
The next Node.js version that will actually allow us to improve the code is v16 which has the web crypto API built-in, and v18 which has it exposed as a global.

~~Still need to test browser support.~~

### CODE REVIEW

**Comment by Richienb** [comment]
@sindresorhus

File: test.js [hunk 1]
```diff
@@ -1,116 +1,139 @@
...
+}
+
+function runTests(test) {
+	// Probabilistic, result is always less than or equal to actual set size, chance it is less is below 1e-256 for sizes up to 32656.
+	const generatedCharacterSetSize = (cryptoRandomString, options, targetSize) => {
+		const set = new Set();
+		const length = targetSize * 640;
+		const string = cryptoRandomString({...options, length});
+
+		for (let i = 0; i < length; i++) {
```
Comment (sindresorhus) [inline test.js:19]:
```suggestion
		for (let index = 0; index < length; index++) {
```

**Comment by sindresorhus** [comment]
Did you test it in the browser? Does the `uvu` thing run in the browser? If so, which one?

File: package.json [hunk 2]
```diff
@@ -17,10 +17,10 @@
 		"browser": "./browser.js"
 	},
 	"engines": {
-		"node": ">=12"
+		"node": ">=14"
```
Comment (sindresorhus) [inline package.json:20]:
```suggestion
		"node": ">=14.16"
```

File: browser.js [hunk 3]
```diff
@@ -1,32 +1,23 @@
...
-	while (byteLength > 0) {
-		const bytesToGenerate = Math.min(byteLength, maxEntropy);
-		generatedBytes.push(crypto.getRandomValues(new Uint8Array({length: bytesToGenerate})));
-		byteLength -= bytesToGenerate;
+	for (let totalGeneratedBytes = 0; totalGeneratedBytes < byteLength; totalGeneratedBytes += maxEntropy) {
+		generatedBytes.set(
+			crypto.getRandomValues(new Uint8Array(Math.min(maxEntropy, byteLength - totalGeneratedBytes))),
+			totalGeneratedBytes,
+		);
 	}
```
[the remark this answers is not in the document]
Comment (Richienb) [reply browser.js:18]: I looked into the source code of the [`randomBytes` polyfill](https://github.com/crypto-browserify/randombytes/blob/f18ded32b209f0d4c637608a11ae042ae96b4c2e/browser.js#L33-L37) and discovered a better way of doing it.

**Comment by Richienb** [comment]
> Did you test it in the browser? Does the `uvu` thing run in the browser? If so, which one?

I switched to `uvu` since AVA only has implicit test suites which are a bit counterproductive since we're trying to run 2 sets of tests which are exactly the same.

To test browser support, the only browser API that is used, `crypto.getRandomValues` is polyfilled with the `crypto.webcrypto.getRandomValues` Node.js API and then the browser code is run in Node.

I've manually verified that the browser code runs in Chrome 100.

**Comment by sindresorhus** [comment]
> I switched to [uvu](https://github.com/lukeed/uvu) since AVA only has implicit test suites which are a bit counterproductive since we're trying to run 2 sets of tests which are exactly the same.

You can still do that with AVA. I know it's not as nice, but I would strongly prefer to use AVA. Not because I made it, but because it's what I use in all my packages. Probably one day the `uvu` thing will become unmaintained and I will have to spend time migrating back to AVA. It has happened so many times.

You can just call AVA twice with different argument and check for that in the test file: https://github.com/avajs/ava/blob/main/docs/recipes/passing-arguments-to-your-test-files.md#passing-arguments-to-your-test-files

AVA may some day get nicer ways to do this kind of thing: https://github.com/avajs/ava/issues/2980 And new issues are always welcome if you can think of a nice way for AVA to handle this. Maybe AVA could have some kind of matrix functionality, like GitHub Actions.

**Comment by Richienb** [comment]
Thought I had seen it used in another pull request that was merged but turns out that it wasn't in one of your repos. Will fix.

**Comment by Richienb** [comment]
@sindresorhus

File: readme.md [hunk 4]
```diff
@@ -48,7 +48,7 @@ cryptoRandomString({length: 10, characters: 'abc'});
 
 Returns a randomized string. [Hex](https://en.wikipedia.org/wiki/Hexadecimal) by default.
 
-### cryptoRandomString.async(options)
+### cryptoRandomStringAsync(options)
```
Comment (sindresorhus) [inline readme.md:51]: We need to indicate somehow that this is a named export and the other one is a default export.

**Comment by sindresorhus** [comment]
@Richienb bump

**Comment by Richienb** [comment]
> We need to indicate somehow that this is a named export and the other one is a default export.

This is quite a challenging task, haven't found inspiration from other repositories yet. My initial idea was to make the first function name italic but it didn't appear visible enough.

**Comment by sindresorhus** [comment]
For now, I think the best way is just a code a code example that includes the import.

**Comment by Richienb** [comment]
Extra sanity check added in Node.js 16.15.0 affected the unit tests, fixed now

**Comment by sindresorhus** [comment]
Nice work

**Pull request merged**